lib/spack/spack/build_systems/autotools.py

```diff
@@ -110,70 +109,49 @@ def _do_patch_config_files(self):
         else:
             config_arch = 'local'
 
-        my_config_files = {'guess': None, 'sub': None}
         config_files = {'guess': None, 'sub': None}
         config_args = {'guess': [], 'sub': [config_arch]}
 
         for config_name in config_files.keys():
             config_file = 'config.{0}'.format(config_name)
-            if os.path.exists(config_file):
-                # First search the top-level source directory
-                my_config_files[config_name] = os.path.abspath(config_file)
-            else:
-                # Then search in all sub directories recursively.
-                # We would like to use AC_CONFIG_AUX_DIR, but not all packages
-                # ship with their configure.in or configure.ac.
-                config_path = next((os.path.abspath(os.path.join(r, f))
-                                    for r, ds, fs in os.walk('.') for f in fs
-                                    if f == config_file), None)
-                my_config_files[config_name] = config_path
-
-            if my_config_files[config_name] is not None:
+            my_config_file_list = find('.', config_file)
```

Comment:
I like using `fs.find` to search for these files, but I would avoid using an hard-coded `.` as the search root. If you do:
```python
fs.find(self.stage.path, files=['config.sub', 'config.guess'], recursive=True)
```
you'll get a list of all the `config.sub` and `config.guess` files in the stage directory and subdirectories:
```
>>> fs.find(s.package.stage.path, files=['config.sub', 'config.guess'], recursive=True)
['/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/config.sub', '/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/config/config.sub', '/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/src/RNAforester/g2-0.72/config.sub', '/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/config.guess', '/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/config/config.guess', '/tmp/culpo/spack-stage/spack-stage-viennarna-2.4.3-aflpwcbnfuoougqztbdzp4xkvz77zy74/spack-src/src/RNAforester/g2-0.72/config.guess']
```
I would use something like that to construct a list of files to be patched.

Reply:
Also, I would rename this list from `my_config_file_list` to something like `to_be_patched`.

Reply:
I create seperate lists for config.guess and config.sub, because These files are copied diffrent files.

Reply:
ping @lee218llnl @alalazo  @xjrc 
I changed to one find function, and check and replace are each suffixes.

Reply:
Apologies @tkameyama Currently swamped with SC20 stuff. I'll try to do my best to get to this before next week, but can't grant it.